app/src/ui/lib/fancy-text-box.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react'
+import { Octicon, OcticonSymbol } from '../octicons'
+import { TextBox, ITextBoxProps } from './text-box'
+import * as classNames from 'classnames'
+
+interface IFancyTextBoxProps extends ITextBoxProps {
+  /** Icon to render */
+  readonly symbol: OcticonSymbol
+
+  /** Callback used to get a reference to internal TextBox */
+  readonly onRef: (textbox: TextBox) => void
+}
+
+interface IFancyTextBoxState {
+  readonly isFocused: boolean
+}
+
+export class FancyTextBox extends React.Component<
```

Comment:
We wanted to extend the `TextBox` component so we wouldn't have to add more _configuration props_. I'm all 👂 on a name if the name is a bit too fancy 🥂 .